File: backend/src/modules/dispatcher/dto/update-dispatcher.dto.ts
import { IsString, IsOptional, IsInt, Min, MaxLength } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';

export class UpdateDispatcherDto {
  @ApiPropertyOptional({ description: '调度组 ID' })
  @IsOptional()
  @IsInt()
  @Min(0)
  setid?: number;

  @ApiPropertyOptional({ description: '目标地址' })
  @IsOptional()
  @IsString()
  @MaxLength(256)
  destination?: string;

  @ApiPropertyOptional({ description: '标志位' })
  @IsOptional()
  @IsInt()
  flags?: number;

  @ApiPropertyOptional({ description: '优先级' })
  @IsOptional()
  @IsInt()
  priority?: number;

  @ApiPropertyOptional({ description: '属性' })
  @IsOptional()
  @IsString()
  @MaxLength(128)
  attrs?: string;

  @ApiPropertyOptional({ description: '描述' })
  @IsOptional()
  @IsString()
  @MaxLength(64)
  description?: string;
}

File: kam-admin-module-routing/src/main/java/io/kamailio/admin/modules/dispatcher/dto/UpdateDispatcherDto.java
package io.kamailio.admin.modules.dispatcher.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Size;
import org.springframework.lang.Nullable;

@Schema(description = "更新调度目标 DTO")
public class UpdateDispatcherDto {

    @Nullable
    @Min(0)
    @Schema(description = "调度组 ID")
    private Integer setid;

    @Nullable
    @Size(max = 256)
    @Schema(description = "目标地址")
    private String destination;

    @Nullable
    @Schema(description = "标志位")
    private Integer flags;

    @Nullable
    @Schema(description = "优先级")
    private Integer priority;

    @Nullable
    @Size(max = 128)
    @Schema(description = "属性")
    private String attrs;

    @Nullable
    @Size(max = 64)
    @Schema(description = "描述")
    private String description;

    public Integer getSetid() { return setid; }
    public void setSetid(Integer setid) { this.setid = setid; }
    public String getDestination() { return destination; }
    public void setDestination(String destination) { this.destination = destination; }
    public Integer getFlags() { return flags; }
    public void setFlags(Integer flags) { this.flags = flags; }
    public Integer getPriority() { return priority; }
    public void setPriority(Integer priority) { this.priority = priority; }
    public String getAttrs() { return attrs; }
    public void setAttrs(String attrs) { this.attrs = attrs; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
}
